### PR TITLE
Interiror grid fix

### PR DESCRIPTION
**Pull request by ChenouZhang**
Now hollow behaves as expected(not creating HOLLOW interior grid anymore).

### CODE REVIEW

**Review by orbeckst — CHANGES_REQUESTED** [review]
Can we make this configurable, i.e., have a setting in the input file or a command line flag? The default should be the old behavior. 

It's very confusing to have tools that behave differently from the original.

**Comment by orbeckst** [comment]
@ChenouZhang can you please address my comments so that we can get this merged?

Also, @rsexton2 might want to use it.

**Comment by ChenouZhang** [comment]
@orbeckst I added flag -i to make it consistent.

File: build/scripts-2.7/hollow [hunk 1]
```diff
@@ -0,0 +1,100 @@
+#!/nfs/homes4/chenou/anaconda3/envs/Born/bin/python
+# Hollow --- http://hollow.sourceforge.net
+# Copyright (c) 2009 Bosco Ho & Franz Gruswitz
+
+from hollow import __version__
+from hollow.core import make_hollow_spheres, defaults, help
```
Comment (orbeckst) [inline build/scripts-2.7/hollow]: I don't think the scripts under build should ever be touched, instead modify the scripts under `scripts`, which are installed.

https://github.com/Becksteinlab/hollow/blob/Interiror-Grid-Fix/scripts/hollow

File: build/scripts-2.7/hollow [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+      help="radius around a grid point, in which the" \
+        + " b-factors of heavy atoms are averaged (0.0=off;" \
+        + " suggested=4.0; default=%.f)" % defaults.bfactor_probe,
+      default=defaults.bfactor_probe)
+  parser.add_option(
+      "-i", dest="is_hollow",  
+      help="Enable/disable the grid.exclude_surrounded " \
+        + "default is yes"  \
+        + "switch to no if one wants interior filled grid",
+      default='yes')
```
Comment (orbeckst) [inline build/scripts-2.7/hollow]: Make it a boolean switch (choose the proper type with `action="store_false"` ) and set the default as `True`.

File: build/scripts-2.7/hollow [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+      default=not defaults.is_skip_waters)
+  parser.add_option(
+      "-b", dest="bfactor_probe", type="float",
+      help="radius around a grid point, in which the" \
+        + " b-factors of heavy atoms are averaged (0.0=off;" \
+        + " suggested=4.0; default=%.f)" % defaults.bfactor_probe,
+      default=defaults.bfactor_probe)
+  parser.add_option(
+      "-i", dest="is_hollow",  
+      help="Enable/disable the grid.exclude_surrounded " \
```
Comment (orbeckst) [inline build/scripts-2.7/hollow]: Describe in words what it does, nobody knows what grid.exclude_surrounded does.

File: hollow/core.py [hunk 4]
```diff
@@ -621,7 +622,15 @@ def toarray(v):
   print "Excluding surrounded points"
   timer.start()
   hole_size = int(1.5 * 1.4 / grid_spacing)
-  grid.exclude_surrounded(hole_size)
+
+
+  print 'is_hollow has been set to: %s' % is_hollow
+  if is_hollow == 'yes':
```
Comment (orbeckst) [inline hollow/core.py]: use a boolean, not string comparison

File: hollow/core.py [hunk 5]
```diff
@@ -621,7 +622,15 @@ def toarray(v):
   print "Excluding surrounded points"
   timer.start()
   hole_size = int(1.5 * 1.4 / grid_spacing)
-  grid.exclude_surrounded(hole_size)
+
+
+  print 'is_hollow has been set to: %s' % is_hollow
```
Comment (orbeckst) [inline hollow/core.py]: A user has no idea what "is_hollow" means. Say in words what happens. 

Maybe refer to the `-i` option of hollow. But think about the messages from the user's perspective.

File: hollow/hollow.txt [hunk 6]
```diff
@@ -6,5 +6,6 @@
   'res_type': 'HOH',
   'atom_type': 'O',
   'atom_field': 'ATOM',
-  'is_skip_waters': True
+  'is_skip_waters': True,
+  'is_hollow': 'yes'
```
Comment (orbeckst) [inline hollow/hollow.txt]: boolean!

**Review by orbeckst — CHANGES_REQUESTED** [review]
Using an optional argument works. Just make it a boolean and see comments on communicating with the user.

EDIT: And importantly: edit the real `hollow` script that is installed.

**Comment by ChenouZhang** [comment]
@orbeckst updated as requested.

File: scripts/hollow [hunk 7]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/nfs/homes4/chenou/anaconda3/envs/Born/bin/python
```
Comment (orbeckst) [inline scripts/hollow]: needs to stay the /usr/bin/env

File: hollow/core.py [hunk 8]
```diff
@@ -621,7 +622,13 @@ def toarray(v):
   print "Excluding surrounded points"
   timer.start()
   hole_size = int(1.5 * 1.4 / grid_spacing)
-  grid.exclude_surrounded(hole_size)
+
+
+  print 'is_hollow has been set to: %r' % is_hollow
```
Comment (orbeckst) [inline hollow/core.py]: see my earlier comment — make it clearer

File: hollow/core.py [hunk 9]
```diff
@@ -621,7 +622,13 @@ def toarray(v):
...
+  if is_hollow:
+    grid.exclude_surrounded(hole_size)
+  else:
+    pass
```
Comment (orbeckst) [inline hollow/core.py]: remove — can just be the if statement without else
```suggestion
```

File: scripts/hollow (hunk 7 above)
Comment (orbeckst) [reply scripts/hollow]:
```suggestion
#!/usr/bin/env python
```

File: hollow/core.py (hunk 8 above)
Comment (orbeckst) [reply hollow/core.py]:
```suggestion
  print 'is_hollow (hollow -i commandline option) has been set to: %r' % is_hollow
```

**Review by orbeckst — APPROVED** [review]
thanks for the changes; I am fixing a few things myself and then merge

**Comment by orbeckst** [comment]
@rsexton2 you should now be able to use `hollow -i` to make it fill the interior with spheres.

**Pull request merged**